lib/puppet/type/esx_vswitch.rb

```diff
@@ -0,0 +1,114 @@
+# Copyright (C) 2013 VMware, Inc.
+Puppet::Type.newtype(:esx_vswitch) do
+  @doc = "Manage vCenter vSwitch."
+
+  ensurable do
+    newvalue(:present) do
+      provider.create
+    end
+    newvalue(:absent) do
+      provider.destroy
+    end
+    defaultto(:present)
+  end
+
+  newparam(:name, :namevar => true) do
+    desc "ESX host:vSwitch name."

```

Comment:
I know this was a port over so it isn't necessary to change this. But I would like to start moving away from this convention of munging properties from a name. It has caused problems in the past. Just taking a moment to call this out.